src/v/kafka/server/server.cc

```diff
@@ -387,10 +393,16 @@ ss::future<response_ptr> sasl_authenticate_handler::handle(
         auto result = co_await ctx.sasl()->authenticate(
           std::move(request.data.auth_bytes));
         if (likely(result)) {
+            ctx.sasl()->set_expiry(ctx.sasl()->credential_expires_in_ms());
+            vlog(
+              klog.warn,
+              "session_lifetime_ms: {}",
+              ctx.sasl()->session_lifetime_ms());
             sasl_authenticate_response_data data{
               .error_code = error_code::none,
               .error_message = std::nullopt,
               .auth_bytes = std::move(result.value()),
+              .session_lifetime_ms = ctx.sasl()->session_lifetime_ms(),
```

Comment:
it's generally preferred to maintain std::chrono types as long as possible, so having `session_lifetime_ms` return a std::chrono::duration and then applying the serialization logic for the response here (ie `std::chrono::duration_cast<ms>(session_lifetime()).count()`)